src/Core/Framework/DataAbstractionLayer/Dbal/EntityWriteGateway.php

```diff
@@ -327,7 +328,9 @@ private function prefetch(EntityDefinition $definition, array $pks, WriteParamet
                 $columns = '(' . $columns . ')';
             }
 
-            $query->andWhere($columns . ' IN (' . $placeholders . ')');
+            if ($params !== []) {
```

Comment:
if there are no primary keys in the definition, this array would be empty, so it would not make sense to add the where condition at all